ar4si/arm-cca.md

```diff
@@ -0,0 +1,91 @@
+# Platform
+
+|Bucket|Notes|Claims|Applicability|
+| --- | --- | --- | --- |
+|Catch-all||VERIFIER_MALFUNCTION|any unexpected exceptions during the appraisal/verification flow|
+|||NO_CLAIM|This MAY be the value for the buckets that are not applicable (i.e., those marked as “N.A.”), if they are included.|
+|||UNEXPECTED_EVIDENCE|On failures coming from the token decoder|
+|||CRYPTO_VALIDATION_FAILED|On failures coming from the Platform token verifier (excluding Realm binding, which is captured in the Realm part)|
+|Instance identity||TRUSTWORTHY_INSTANCE|verification using CPAK is successful and Implementation ID is known (implicitly, from the fact that ref-values associated to it exist)|
+|||UNTRUSTWORTHY_INSTANCE|If the verifier has a notion of known-bad CPAK, then verification using CPAK that is successful but CPAK is known-bad can assert this claim.|
+|||UNRECOGNIZED_INSTANCE|No CPAK found corresponding to the Instance ID or Implementation ID is not known (i.e., there are no ref-values associated to it.)|
+|Config|Two notions of config in CCA may apply: (a) the “System Properties” captured in the Platform “config” claim, (b) the firmware and/or hardware configuration file that can be specified by certain boot loader stages. Those end up in sw-components entries of type “*_CONFIG”|APPROVED_CONFIG|TBD[^1] decide on what semantics we want to map here.|
+|||NO_CONFIG_VULNS|ditto|
+|||UNAVAIL_CONFIG_ELEMS|ditto|
+|||UNSAFE_CONFIG|ditto|
+|||UNSUPPORTABLE_CONFIG|ditto|
+|Executables|This conflates boot and run-time and makes disentangling “partial match” situations impossible.|APPROVED_BOOT|If all sw-component claims have been verified|
+|||APPROVED_RUNTIME|N.A.|
+|||CONTRAINDICATED_RUNTIME|N.A.|
+|||UNRECOGNIZED_RUNTIME|N.A.|
+|||UNSAFE_RUNTIME|N.A.|
+|File system||APPROVED_FILES|N.A.|
+|||UNRECOGNIZED_FILES|N.A.|
+|||CONTRAINDICATED_FILES|N.A.|
+|Hardware (and Firmware)||CONTRAINDICATED_HARDWARE (genuine HW/FW but contraindicated)|TRUSTWORTHY_INSTANCE && some sw-component claims are known-bad|
+|||UNRECOGNIZED_HARDWARE (unrecognised HW/FW)|TRUSTWORTHY_INSTANCE && some sw-component claims couldn’t be found|
+|||GENUINE_HARDWARE (genuine HW/FW)|Same as (TRUSTWORTHY_INSTANCE && APPROVED_BOOT)|
+||Not clear what is the difference between this and CONTRAINDICATED_HARDWARE?|UNSAFE_HARDWARE (genuine HW/SW but known sec vulns)|TBD (see note)|
```

Comment:
>Not clear what is the difference between this and CONTRAINDICATED_HARDWARE?

CONTRAINDICATED ==  know exploits
UNSAFE == known vulns (but no know exploits confirmed)
?

Reply:
I like the idea of discriminating based on the exploitability of the vulnerability.
However, I'd flip the assignments.  UNSAFE is 32, i.e., a warning, while CONTRAINDICATED is 96, i.e., at a higher level.

Reply:
Note for self: this needs a corresponding signal in the x-refval.